src/m1_facilitator/Manifest/Manifest.ts

```diff
@@ -0,0 +1,259 @@
+ * It holds avatar information.
+ */
+export class Avatar {
+  public readonly keystorePath: string;
+
+  public readonly keystorePasswordPath: string;
+
+  /**
+   * Constructor.
+   *
+   * @param keystorePath File path containing keystore path.
+   * @param keystorePasswordPath File path containing keystore password.
+   */
+  public constructor(keystorePath: string, keystorePasswordPath: string) {
+    this.keystorePath = keystorePath;
+    this.keystorePasswordPath = keystorePasswordPath;
+  }
+}
+
+/**
+ * It holds chain and graph information.
+ */
+export class Chain {
+  /** Chain rpc/ws endpoint. */
+  public readonly nodeEndpoint: string;
+
+  /** Subgraph web socket endpoint */
+  public readonly graphWsEndpoint: string;
+
+  /** Subgraph rpc endpoint */
+  public readonly graphRpcEndpoint: string;
+
+  /** Avatar account address. */
+  public readonly avatarAccount: string;
+
+  /**
+   * Constructor
+   *
+   * @param nodeEndpoint Chain end point.
+   * @param graphWsEndpoint Graph web socket end point.
+   * @param graphRpcEndpoint Graph rpc end point.
+   * @param avatarAccount Avatar account address.
+   */
+  public constructor(
+    nodeEndpoint: string,
+    graphWsEndpoint: string,
+    graphRpcEndpoint: string,
+    avatarAccount: string,
+  ) {
+    this.nodeEndpoint = nodeEndpoint;
+    this.graphWsEndpoint = graphWsEndpoint;
+    this.graphRpcEndpoint = graphRpcEndpoint;
+    this.avatarAccount = avatarAccount;
+  }
+}
+
+/**
+ * Represent metachain which contains origin and auxiliary chain objects.
+ */
+export class Metachain {
+  public readonly originChain: Chain;
+
+  public readonly auxiliaryChain: Chain;
+
+  /**
+   * Constructor.
+   *
+   * @param originChain Origin chain object.
+   * @param auxiliaryChain Auxiliary chain object.
+   */
+  public constructor(originChain: Chain, auxiliaryChain: Chain) {
+    this.originChain = originChain;
+    this.auxiliaryChain = auxiliaryChain;
+  }
+}
+
+/**
+ * The object represents facilitator manifest values.
+ */
+export default class Manifest {
+  public readonly version: string;
+
+  public readonly architectureLayout: ArchitectureLayout;
+
+  public readonly personas: Personas[];
+
+  public readonly metachain: Metachain;
+
+  public readonly dbConfig: DBConfig;
+
+  public readonly avatars: Record<string, Avatar>;
+
+  public readonly originContractAddresses: Record<string, string>;
+
+  public readonly facilitateTokens: Set<string>;
+
+  /**
+   * Constructor.
+   *
+   * @param config Facilitator input config object.
+   */
+  private constructor(config: {
+    version: string;
+    architecture_layout: string;
+    personas: string[];
+    metachain: Metachain;
+    accounts: Record<string, Avatar>;
+    origin_contract_addresses: Record<string, string>;
+    facilitate_tokens: string[];
+  }) {
+    this.version = config.version;
+    this.architectureLayout = config.architecture_layout as ArchitectureLayout;
+    this.personas = config.personas as Personas[];
```

Comment:
Question: This will type cast cash each  value of array?

Reply:
Yes!